Stratis.Bitcoin.Features.Miner/MiningFeature.cs

```diff
@@ -67,6 +68,29 @@ public class MiningFeature : FullNodeFeature
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
         }
 
+        /// <summary>
+        /// Starts staking a wallet.
+        /// </summary>
+        /// <param name="walletName">The name of the wallet.</param>
+        /// <param name="walletPassword">The password of the wallet.</param>
+        public void StartStaking(string walletName, string walletPassword)
```

Comment:
Ho I just realised what if this is called more then once?
Maybe like this
```
if(this.posLoop != null)
   return
```

Well lets commit and fix this in another PR

Reply:
This code already guarantees that the current "this.posLoop" will be re-used if we are already mining - essentially having the same effect:

        public IAsyncLoop Mine(WalletSecret walletSecret)
        {
            this.logger.LogTrace("()");
            if (this.mining != null)
            {
                this.logger.LogTrace("(-)[ALREADY_MINING]");
                return this.mining;
            }

Reply:
@quantumagi is right, there is usually no problem if this is called twice except for very corner case of very quick double call during the very first time initialization, in which case this code that @quantumagi copied here is prone to race condition where mining will be null and the if branch won't be executed and if meanwhile second thread calls Mine, it would create two async loops. But this is very very unlikely to happen, almost requires a user's intention to make a mess. So not sure if it is worth of handling.